downstreamadapter/sink/redo/meta.go

```diff
@@ -0,0 +1,539 @@
+		return false
+	}
+
+	commitTs, fileType, err := redo.ParseLogFileName(path)
+	if err != nil {
+		log.Error("parse file name failed",
+			zap.String("namespace", m.changeFeedID.Namespace()),
+			zap.String("changefeed", m.changeFeedID.Name()),
+			zap.String("path", path), zap.Error(err))
+		return false
+	}
+	if fileType != redo.RedoDDLLogFileType && fileType != redo.RedoRowLogFileType {
+		log.Panic("unknown file type",
+			zap.String("namespace", m.changeFeedID.Namespace()),
+			zap.String("changefeed", m.changeFeedID.Name()),
+			zap.String("path", path), zap.Any("fileType", fileType))
+	}
+
+	// if commitTs == checkPointTs, the DDL may be executed in the owner,
+	// so we should not delete it.
+	return commitTs < checkPointTs
+}
+
+// deleteAllLogs delete all redo logs and leave a deleted mark.
+func (m *RedoMeta) deleteAllLogs(ctx context.Context) error {
+	// when one changefeed with redo enabled gets deleted, it's extStorage should always be set to not nil
+	// otherwise it should have already meet panic during changefeed running time.
+	// the extStorage may be nil in the unit test, so just set the external storage to make unit test happy.
+	if m.extStorage == nil {
+		uri, err := storage.ParseRawURL(m.cfg.Storage)
+		redo.FixLocalScheme(uri)
+		if err != nil {
+			return err
+		}
+		m.extStorage, err = redo.InitExternalStorage(ctx, *uri)
+		if err != nil {
+			return err
+		}
+	}
+	// Write deleted mark before clean any files.
+	deleteMarker := getDeletedChangefeedMarker(m.changeFeedID)
+	if err := m.extStorage.WriteFile(ctx, deleteMarker, []byte("D")); err != nil {
+		return errors.WrapError(errors.ErrExternalStorageAPI, err)
+	}
+	log.Info("redo manager write deleted mark",
+		zap.String("namespace", m.changeFeedID.Namespace()),
+		zap.String("changefeed", m.changeFeedID.Name()))
+
+	changefeedMatcher := getChangefeedMatcher(m.changeFeedID)
+	return util.RemoveFilesIf(ctx, m.extStorage, func(path string) bool {
+		if path == deleteMarker || !strings.Contains(path, changefeedMatcher) {
+			return false
+		}
+		return true
+	}, nil)
+}
+
+func (m *RedoMeta) maybeFlushMeta(ctx context.Context) error {
+	hasChange, unflushed := m.prepareForFlushMeta()
+	if !hasChange {
+		// check stuck
+		if time.Since(m.lastFlushTime) > redo.FlushWarnDuration {
+			log.Debug("Redo meta has not changed for a long time, owner may be stuck",
+				zap.String("namespace", m.changeFeedID.Namespace()),
+				zap.String("changefeed", m.changeFeedID.Name()),
+				zap.Duration("lastFlushTime", time.Since(m.lastFlushTime)),
+				zap.Any("meta", unflushed))
+		}
+		return nil
+	}
+
+	log.Debug("Flush redo meta",
+		zap.String("namespace", m.changeFeedID.Namespace()),
+		zap.String("changefeed", m.changeFeedID.Name()),
+		zap.Any("meta", unflushed))
+	if err := m.flush(ctx, unflushed); err != nil {
+		return err
+	}
+	m.postFlushMeta(unflushed)
+	m.lastFlushTime = time.Now()
+	return nil
+}
+
+func (m *RedoMeta) prepareForFlushMeta() (bool, misc.LogMeta) {
```

Comment:
I'm always confused with this values of flushedxxxx, unflushedxxx. Could you please give a clear comment here to show what these values mean.